src/vmm_mad/remotes/kvm/poll

```diff
@@ -396,6 +396,30 @@ module KVM
         end
     end
 
+    def self.get_io_statistics(vmid, text = nil)
+        #disks=get_disk_names(vmid, text)
+        #puts "disk = #{disks}"
+        vm_stats=`#{virsh(:domstats)} #{vmid}`
+
+        #if disks && !disks.empty?
+        values={}
+        values[:netrx] = 0
+        values[:nettx] = 0
+        values[:diskrdbytes]=0
+        values[:diskwrbytes]=0
+        values[:diskrdiops]=0
+        values[:diskwriops]=0
+
+        vm_stats.scan(/rx.bytes=(\d*)/).each { |value| values[:netrx]+=value[0].to_i }
```

Comment:
If this is all is for the performance reasons, it would be better to have just a single pass over the `vm_stats` (similar way as old methods `get_diskio_statistics` etc.).

Also, regular expression better with at least one number `(\d+)`.